hoodie-client/src/test/java/com/uber/hoodie/table/TestMergeOnReadTable.java

```diff
@@ -564,6 +567,160 @@ public void testRollbackWithDeltaAndCompactionCommit() throws Exception {
+        metaClient.getCommitTimeline().filterCompletedInstants(), allFiles);
+    Stream<HoodieDataFile> dataFilesToRead = roView.getLatestDataFiles();
+    assertTrue(!dataFilesToRead.findAny().isPresent());
+
+    roView = new HoodieTableFileSystemView(metaClient, hoodieTable.getCompletedCommitTimeline(), allFiles);
+    dataFilesToRead = roView.getLatestDataFiles();
+    assertTrue("ReadOptimizedTableView should list the parquet files we wrote in the delta commit",
+        dataFilesToRead.findAny().isPresent());
+
+    /**
+     * Write 2 (inserts + updates)
+     */
+    newCommitTime = "002";
+    allCommits.add(newCommitTime);
+    // WriteClient with custom config (disable small file handling)
+    HoodieWriteClient nClient = new HoodieWriteClient(jsc, HoodieWriteConfig.newBuilder().withPath(basePath)
+        .withSchema(TRIP_EXAMPLE_SCHEMA)
+        .withParallelism(2, 2)
+        .withAutoCommit(false).withAssumeDatePartitioning(true).withCompactionConfig(HoodieCompactionConfig.newBuilder()
+            .compactionSmallFileSize(1 * 1024).withInlineCompaction(false)
+            .withMaxNumDeltaCommitsBeforeCompaction(1).build())
+        .withStorageConfig(HoodieStorageConfig.newBuilder().limitFileSize(1 * 1024).build())
+        .forTable("test-trip-table").build());
+    nClient.startCommitWithTime(newCommitTime);
+
+    List<HoodieRecord> copyOfRecords = new ArrayList<>(records);
+    copyOfRecords = dataGen.generateUpdates(newCommitTime, copyOfRecords);
+    copyOfRecords.addAll(dataGen.generateInserts(newCommitTime, 200));
+
+    List<String> dataFiles = roView.getLatestDataFiles().map(hf -> hf.getPath()).collect(Collectors.toList());
+    List<GenericRecord> recordsRead = HoodieMergeOnReadTestUtils.getRecordsUsingInputFormat(dataFiles, basePath);
+    assertEquals(recordsRead.size(), 200);
+
+    statuses = nClient.upsert(jsc.parallelize(copyOfRecords, 1), newCommitTime).collect();
+    // Verify there are no errors
+    assertNoWriteErrors(statuses);
+    nClient.commit(newCommitTime, writeStatusJavaRDD);
+    copyOfRecords.clear();
+
+    // Compaction commit
+    /**
+     * Write 3 (updates)
+     */
+    newCommitTime = "003";
+    allCommits.add(newCommitTime);
+    client.startCommitWithTime(newCommitTime);
+
+    records = dataGen.generateUpdates(newCommitTime, records);
+
+    writeStatusJavaRDD = client.upsert(writeRecords, newCommitTime);
+    client.commit(newCommitTime, writeStatusJavaRDD);
+    statuses = writeStatusJavaRDD.collect();
+    // Verify there are no errors
+    assertNoWriteErrors(statuses);
+
+    metaClient = new HoodieTableMetaClient(jsc.hadoopConfiguration(), cfg.getBasePath());
+
+    String compactionInstantTime = "004";
+    allCommits.add(compactionInstantTime);
+    client.scheduleCompactionAtInstant(compactionInstantTime, Optional.empty());
+    JavaRDD<WriteStatus> ws = client.compact(compactionInstantTime);
+    client.commitCompaction(compactionInstantTime, ws, Optional.empty());
```

Comment:
Can we also have a pending compaction covered in the test-case

Reply:
Done